src/main/java/Objects/NetworkDevices/NetworkDevice.java

```diff
@@ -19,13 +28,92 @@ public String getId() {
         return id;
     }
 
+    /**
+     * Sets the types of packets each device can send.
+     * @param packet_type The types of packets each device can send.
+     */
+    public void setPacket(final String packet_type) {
```

Comment:
I felt this naming convention was better, because you'd use this when setting it (for example, you wouldn't take a device that can't send botnet packets and then add the ability for it to do so).
Doing the list way would include creating a list, instead of running a few add commands.  I don't see that is a better way of doing.

Reply:
So we would never be adding a packet to a device that it didn't already have correct? In that case we really should just be setting the list.

Reply:
unresolved, this is the same thing as setting connections, could be done easily using Arrays.asList